### PR TITLE
feature/erc20-library

### PR DESCRIPTION
**Pull request by mattdf**

### CODE REVIEW

**Review by Arachnid — APPROVED** [review]

File: test/erc20.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  before(async () => {
+    erc20 = await deployLibrary("ERC20Ops");
+    
+    eventsContract = await (await ethers.getContractFactory("Events")).deploy();
+    events = weiroll.Contract.fromEthersContract(eventsContract);
+
+    /* Deploy token contract */
+    tokenContract = await (await ethers.getContractFactory("ExecutorToken")).deploy(supply);
+
+    /* transfer some balance to executor since we're not delegatecalling currently */
```
Comment (Arachnid) [inline test/erc20.js:28]: This comment doesn't reflect what's going on here.

**Review by Arachnid — APPROVED** [review]

File: contracts/Libraries/ERC20.sol [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external returns (bool);
+
+    event Transfer(address indexed from, address indexed to, uint256 value);
+
+    event Approval(
```
Comment (decanus) [inline contracts/Libraries/ERC20.sol:27]: @mattdf i think nows a good time to import openzeppelin, a lot of our libs will use interfaces they have declared already. Wdyt @Arachnid

Comment (mattdf) [reply contracts/Libraries/ERC20.sol:27]: I did import it for the actual ERC20 token, I just didn't want to depend on it for the interface to keep it more self contained.

Comment (decanus) [reply contracts/Libraries/ERC20.sol:27]: I'd say this is a good part to import the interfaces, easier for auditors to know we have the right one etc.

Comment (mattdf) [reply contracts/Libraries/ERC20.sol]: kk, i replaced it with an import from zep

Comment (Arachnid) [reply contracts/Libraries/ERC20.sol]: Agreed, we should use the OZ interfaces wherever practical.

**Review by Arachnid — APPROVED** [review]

**Review by Arachnid — APPROVED** [review]

**Pull request merged**